apps/src/componentLibrary/dropdown/iconDropdown/IconDropdown.tsx

```diff
@@ -0,0 +1,142 @@
+import React, {useCallback, memo} from 'react';
+import classNames from 'classnames';
+
+import {
+  ComponentSizeXSToL,
+  DropdownColor,
+} from '@cdo/apps/componentLibrary/common/types';
+import moduleStyles from '@cdo/apps/componentLibrary/dropdown/customDropdown.module.scss';
+
+import CustomDropdown from '@cdo/apps/componentLibrary/dropdown/_CustomDropdown';
+
+import FontAwesomeV6Icon, {
+  FontAwesomeV6IconProps,
+} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+
+import {dropdownColors} from '@cdo/apps/componentLibrary/common/constants';
+import {
+  DropdownProviderWrapper,
+  useDropdownContext,
+} from '@cdo/apps/componentLibrary/common/contexts/DropdownContext';
+
+export interface IconDropdownOption {
+  value: string;
+  label: string;
+  isOptionDisabled?: boolean;
+  icon: FontAwesomeV6IconProps;
+}
+
+export interface IconDropdownProps {
+  /** IconDropdown name */
```

Comment:
Nit: do these comments get used somehow? Feels like it makes it harder to read / adds little new information.

Reply:
+1 to Ben, nice to have these in Storybook but it would be good to add a little more explanation where possible instead of just restating the field name? For example, something like - `name: "Name of the dropdown, used for uniquely identifying the dropdown's HTML elements"` and `labelText: "The user-facing label of the dropdown"` - would help explain the difference between these two properties. Can definitely be a follow-up though